source/agora/test/NetworkManager.d

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+
+    Contains tests for the tests & error-handling of the NetworkManager
+
+    Copyright:
+        Copyright (c) 2019 BOS Platform Foundation Korea
+        All rights reserved.
+
+    License:
+        MIT License. See LICENSE for details.
+
+*******************************************************************************/
+
+module agora.test.NetworkManager;
+
+version (unittest):
+
+import agora.common.crypto.Key;
+import agora.common.Block;
+import agora.common.Data;
+import agora.common.Hash;
+import agora.common.Transaction;
+import agora.consensus.Genesis;
+import agora.test.Base;
+
+/// test behavior when getBlockHeight() call fails
+unittest
+{
+    import std.algorithm;
+    import std.range;
+    import core.thread;
+
+    const NodeCount = 4;
+    auto network = makeTestNetwork!TestNetworkManager(NetworkTopology.Simple, NodeCount);
+    network.start();
+    assert(network.getDiscoveredNodes().length == NodeCount);
+
+    auto nodes = network.apis.values;
+    auto node_1 = nodes[0];
+
+    // first two nodes will fail, second two should work
+    nodes[0].filter!(node_1.getBlockHeight);
+    nodes[1].filter!(node_1.getBlockHeight);
```

Comment:
Style nit: Can we go with `API.getBlockHeight` instead of `node_1.gitBlockHeight` ? The former makes it more obvious that the instance is irrelevant.